### PR TITLE
Clean autocoders code

### PR DESCRIPTION
**Pull request by iw4p**
| | |
|:---|:---|
|**_Originating Project/Creator_**| iw4p |
|**_Affected Component_**| Autocoder  |
|**_Affected Architectures(s)_**| Python |
|**_Related Issue(s)_**| N/A |
|**_Has Unit Tests (y/n)_**| exist |
|**_Builds Without Errors (y/n)_**| Y |
|**_Unit Tests Pass (y/n)_**| Y |
|**_Documentation Included (y/n)_**| Y |

---
## Change Description
Check and lint codes via pylint to find out what's missing or wrong and clean the code.

## Rationale
Fixed bugs.

## Testing/Review Recommendations

Built with `cmake ../ -DCMAKE_BUILD_TYPE=TESTING` and `make check`.

## Future Work

Nothing for now.

### CODE REVIEW

File: Autocoders/Python/bin/gds_dictgen.py [hunk 1]
```diff
@@ -192,12 +192,10 @@ def main():
     #
     if len(args) == 0:
         print("Usage: %s [options] xml_filename" % sys.argv[0])
-        return
```
Comment (LeStarch) [inline Autocoders/Python/bin/gds_dictgen.py:195]: Return is necessary here.

**Review by LeStarch — CHANGES_REQUESTED** [review]
Many of the returns that were removed were necessary for the code to function.  In all cases where a return is on an error branch they need to be restored.

File: Autocoders/Python/bin/gds_dictgen.py (hunk 1 above)
Comment (iw4p) [reply Autocoders/Python/bin/gds_dictgen.py]: Oh, I think it's fine now.

File: Autocoders/Python/bin/implgen.py [hunk 2]
```diff
@@ -218,12 +218,10 @@ def main():
     #
     if len(args) == 0:
         print("ERROR: Usage: %s [options] xml_filename" % sys.argv[0])
-        return
```
Comment (LeStarch) [inline Autocoders/Python/bin/implgen.py:221]: This return is still removed, which will change the behavior

**Comment by iw4p** [comment]
> Two more missing returns. Other than that, it looks good!

I think it's fine now, Thank you.

**Review by LeStarch — APPROVED** [review]

**Pull request merged**